zerver/openapi/zulip.yaml

```diff
@@ -14958,19 +14950,30 @@ paths:
                               permission to create public channels.
                       realm_create_public_stream_policy:
                         type: integer
+                        deprecated: true
                         description: |
                           Present if `realm` is present in `fetch_event_types`.
 
-                          The [policy][permission-level] for which users can create public channels
-                          in this organization.
+                          This is set based on the value of `can_create_public_channel_group`
+                          setting to control who can create public channels.
+
+                          It is one of the below values if the group setting is set to one
+                          of the corresponding system user groups -
 
                           - 1 = Members only
                           - 2 = Admins only
                           - 3 = [Full members][calc-full-member] only
                           - 4 = Admins and moderators only
 
-                          **Changes**: Before Zulip 5.0 (feature level 102), permission to
-                          create channels was controlled by the `realm_create_stream_policy` setting.
+                          If the `can_create_public_channel_group` setting is not set to
+                          one of the corresponding system groups, this value is set to the
+                          value representing "Members only" option.
```

Comment:
I think we can be a bit more clear here about the recommendation for clients, something like

> `realm_create_public_stream_policy` is a deprecated representation of a superset of the users who have permission to create public channels in the organization, available for backwards-compatibility. Clients should use `can_create_public_channel_group` instead.

> It is an enum with the following possible values, corresponding to roles/system groups: